### PR TITLE
Add framework and external to common binary 

### PR DESCRIPTION
**Pull request by abhishek-samsung**
Added framework and external to common binary. Task Manager shares a variable with apps/builtin to register the builtin
apps information such as the entry point. With commit "Adding Framework and External to Common Binary", calls from Framework to Apps is not be possible. To resolve this issue, adding an API in Task Manager to initialize the above shared variables. This API is called just after the Task Manager is launched.

### CODE REVIEW

File: framework/src/task_manager/task_manager_core.c [hunk 1]
```diff
@@ -1495,6 +1500,20 @@ static int taskmgr_init_task_manager(void)
 	return OK;
 }
 
+void task_manager_populate_builtin_list(const builtin_info_t builtin_list[], int builtin_list_count)
+{
+	tm_builtin_list = TM_ALLOC(builtin_list_count * sizeof(tm_builtin_info_t));
```
Comment (dineshdas-samsung) [inline framework/src/task_manager/task_manager_core.c:1505]: Can we please check tm_builtin_list is null ?

**Review by deepaksrma — COMMENTED** [review]
change the commit msg like this:

`framework/task_manager: Fix for Framework accessing App side code`

because we need to tell in the commit where we are making changes.

same for other commit.
and 1 question, why did you split the commit?

File: framework/include/task_manager/task_manager.h [hunk 2]
```diff
@@ -102,6 +103,21 @@ enum tm_defined_broadcast_msg {
 	TM_BROADCAST_MSG_MAX = TM_BROADCAST_SYSTEM_MSG_MAX,
 #endif
 };
+/**
```
Comment (amogh-samsung) [inline framework/include/task_manager/task_manager.h:106]: Could you please leave a line's space after the previous structure definition?

File: framework/src/task_manager/task_manager_core.c [hunk 3]
```diff
@@ -1495,6 +1500,20 @@ static int taskmgr_init_task_manager(void)
...
+	for (int i = 0; i < builtin_list_count; i++) {
+		strncpy(tm_builtin_list[i].name, builtin_list[i].name, CONFIG_TASK_NAME_SIZE);
+		tm_builtin_list[i].entry = builtin_list[i].entry;
+		tm_builtin_list[i].exectype = builtin_list[i].exectype;
+		tm_builtin_list[i].priority = builtin_list[i].priority;
+		tm_builtin_list[i].stacksize = builtin_list[i].stacksize;
+	}
+
+	builtin_list_cnt = builtin_list_count;
```
Comment (amogh-samsung) [inline framework/src/task_manager/task_manager_core.c:1514]: Possible to make these two variable names a bit more distinct?
"cnt" vs "count" could be confusing in the future.

File: framework/src/task_manager/task_manager_permission.c [hunk 4]
```diff
@@ -32,7 +31,7 @@ bool taskmgr_is_permitted(int handle, pid_t pid)
 	int builtin_cnt;
 	int permission = TM_PERMISSION(handle);
 
-	builtin_cnt = get_builtin_list_cnt();
+	builtin_cnt = builtin_list_cnt;
```
Comment (amogh-samsung) [inline framework/src/task_manager/task_manager_permission.c:34]: Similarly, here as well.
The two variable names in use are fairly close to each other.

**Review by amogh-samsung — COMMENTED** [review]
Could you shorten the commit description in addition to the redefining of the commit message as suggested by @deepaksrma 
My suggestion would be -

> -Task Manager shares a variable to register the builtin apps information.
> -However, calls from Framework to Apps are not yet possible.
> -Hence, add an API in Task Manager to initialize the shared variable.
> -This API is called just after the Task Manager is launched.

You can decide better based on the context.

**Comment by jeongchanKim** [comment]
I understand why you changed like this.
But I think, with loadable configuration, apps folder can be into common binary also.
Because real user apps will be in loadable_apps, and the apps in apps folder are almost about system utils.
(For example, TASH is in apps folder, and it should be into common binary.)
How do you think about?

**Comment by kishore-sn** [comment]
@jeongchanKim 
If apps can be moved into common binary, then this change wont be necessary. 
Since, we have been keeping the apps in the app binary till now, we thought of making this change.

File: framework/src/task_manager/task_manager_core.c [hunk 5]
```diff
@@ -1495,6 +1500,20 @@ static int taskmgr_init_task_manager(void)
 	return OK;
 }
 
+void task_manager_populate_builtin_list(const builtin_info_t builtin_list[], int builtin_list_count)
+{
+	tm_builtin_list = TM_ALLOC(builtin_list_count * sizeof(tm_builtin_info_t));
+	for (int i = 0; i < builtin_list_count; i++) {
```
Comment (thapav) [inline framework/src/task_manager/task_manager_core.c:1506]: declare i in the start of the function separately

File: framework/src/task_manager/task_manager_core.c (hunk 1 above)
Comment (Abhishek-05) [reply framework/src/task_manager/task_manager_core.c:1505]: Done

File: framework/src/task_manager/task_manager_core.c (hunk 5 above)
Comment (Abhishek-05) [reply framework/src/task_manager/task_manager_core.c:1506]: Done

File: framework/src/task_manager/task_manager_core.c (hunk 3 above)
Comment (Abhishek-05) [reply framework/src/task_manager/task_manager_core.c:1514]: Changed the global variable name to tm_builtin_list_cnt

File: framework/src/task_manager/task_manager_permission.c (hunk 4 above)
Comment (Abhishek-05) [reply framework/src/task_manager/task_manager_permission.c:34]: Done

File: framework/include/task_manager/task_manager.h (hunk 2 above)
Comment (Abhishek-05) [reply framework/include/task_manager/task_manager.h:106]: Done

File: framework/src/task_manager/task_manager_core.c (hunk 1 above)
Comment (abhishek-samsung) [reply framework/src/task_manager/task_manager_core.c]: Done.

File: framework/include/task_manager/task_manager.h (hunk 2 above)
Comment (abhishek-samsung) [reply framework/include/task_manager/task_manager.h:107]: Done.

File: framework/src/task_manager/task_manager_core.c (hunk 3 above)
Comment (abhishek-samsung) [reply framework/src/task_manager/task_manager_core.c]: Changed the global variable name to tm_builtin_list_cnt .

File: framework/src/task_manager/task_manager_permission.c (hunk 4 above)
Comment (abhishek-samsung) [reply framework/src/task_manager/task_manager_permission.c]: Done.

File: framework/src/task_manager/task_manager_core.c (hunk 5 above)
Comment (abhishek-samsung) [reply framework/src/task_manager/task_manager_core.c]: Done.

**Comment by jeongchanKim** [comment]
> If apps can be moved into common binary, then this change wont be necessary.
> Since, we have been keeping the apps in the app binary till now, we thought of making this change.

Yes, I know.
I discussed with @sunghan-chang that apps folder can be into common binary. 
The reason is what I said before.
If you have another opinion, please let me know.

**Pull request closed without merging**